website/blog/2023-04-18-LLM-tuning-math/index.mdx

```diff
@@ -0,0 +1,68 @@
+
+In this blog post, we will explore how model and inference parameter matter in LLM applications, using a case study for [MATH](https://datasets-benchmarks-proceedings.neurips.cc/paper/2021/hash/be83ab3ecd0db773eb2dc1b0a17836a1-Abstract-round2.html), a benchmark for evaluating LLMs on advanced mathematical problem solving. MATH consists of 12K math competition problems from AMC-10, AMC-12 and AIME. Each problem is accompanied by a step-by-step solution.
+
+We will use the new subpackage [`flaml.autogen`](docs/Use-Cases/Auto-Generation) to automatically find the best model and inference parameter for LLMs on a given task and dataset given an inference budget, using a novel low-cost search & pruning strategy. FLAML currently supports all the LLMs from OpenAI, such as GPT-3.5 and GPT-4.
+
+We will use FLAML to perform model selection and inference parameter tuning. Then we compare the performance and inference cost on solving algebra problems with the untuned gpt-4. We will also analyze how different difficulty levels affect the results.
+
+## Experiment Setup
+
+We use FLAML to select between the following models with a target inference budget $0.02 per instance:
+- gpt-3.5-turbo, a relatively cheap model that powers the popular ChatGPT app
+- gpt-4, the state of the art LLM that costs more than 100 times of gpt-3.5-turbo
+
+We adapt the models using 20 examples in the train set, using the problem statement as the input and generating the solution as the output. We use the following inference parameters:
+
+- temperature: The parameter that controls the randomness of the output text. A higher temperature means more diversity but less coherence. We search for the optimal temperature in the range of [0, 1].
+- top_p: The parameter that controls the probability mass of the output tokens. Only tokens with a cumulative probability less than or equal to top-p are considered. A lower top-p means more diversity but less coherence. We search for the optimal top-p in the range of [0, 1].
+- max_tokens: The maximum number of tokens that can be generated for each output. We search for the optimal max length in the range of [50, 1000].
+- n: The number of responses to generate. Generating multiple responses and select the best among them can help with resolving potential errors due to randomness. In this experiment, we find the answer with highest votes among all the responses and then select it as the final answer to compare with the ground truth. We search for the optimal n in the range of [1, 100].
+- prompt: We use the template: "{problem} Solve the problem carefully. Simplify your answer as much as possible. Put the final answer in \\boxed{{}}." where {problem} will be replaced by the math problem instance.
+
+We use the average accuracy and inference cost as the metric to evaluate the performance. The inference cost is measured by the price per 1K tokens and the number of tokens consumed.
```

Comment:
Explain how to evaluate whether a problem is solved when there are multiple responses?

Reply:
Do you mean to expand "In this experiment, we find the answer with highest votes among all the responses and then select it as the final answer to compare with the ground truth"?

Reply:
I also think it helps to explain how choosing an answer from multiple answers could help improve the score.

For example, can we show the top-n outputs by GPT?

Reply:
There are 54 responses from gpt-3.5-turbo. Do you want to see a toy example? e.g.,

"For example, if n=5 and 3 of the responses contain a final answer \boxed{301} while 2 of the responses contain a final answer \boxed{159}, we chose 301 as the final answer."